Framework/API/src/DetectorInfo.cpp

```diff
@@ -93,6 +93,20 @@ Kernel::Quat DetectorInfo::rotation(const size_t index) const {
   return getDetector(index).getRotation();
 }
 
+double DetectorInfo::eFixed(const size_t index) const {
+  const auto &detector = getDetector(index);
+  auto eFixedVector =
+      detector.getNumberParameter(Geometry::ParameterMap::eFixed());
+
+  double eFixed = 0.0;
+
+  if (!eFixedVector.empty()) {
+    eFixed = eFixedVector.front();
+  }
+
```

Comment:
@SimonHeybrock. These errors could be ignored. However, as @martyngigg pointed out its up to the client to decide if the absence of one, or more, or these parameters is really an error or not. I would say that given that the client of this code is an Algorithm author, they should be responsible for testing these scenarios anyway at the algorithm level.